### DISCUSSION TITLE
Notifications are very unreliable while app is open, even if in background and the phone is locked

### DISCUSSION THREAD

**Discussion by pct-cclausen** [capawesome-team/capacitor-firebase#541] [Q&A] [answered]
We're struggeling with android push notifications:

- In many cases our users end up with the app open (or maybe in background) and the phone locked
- In these cases what seems to happen is that the push messages are not shown, instead they are given to the app
- The app then processes them only once the phone is unlocked, which means if the user does not unlock their phone there is no "ping" notification showing up for them at all

We need notifications to always show in the notification center. I tried to create a local notification when a notification is received while in foreground, but that code seems to only run once the phone is unlocked and users may not do that until it is too late.

The entire "do something different when app is open or in foreground"-codepath is a problem to us, I'd like to disable it completely. Things work great as long as the app does not hit that codepath.

**Comment by robingenz** [comment]
Push notifications are definitely delivered and displayed, even if the phone is locked. The requirement is, of course, that the permission has been granted and that the plugin has been configured correctly.

**Reply by pct-cclausen** [reply]
If the app is closed they are displayed while the phone is locked.

But:
1. Open the app
2. Lock the phone with the app open
3. A notification is received, nothing is shown
4. Unlock the phone: Now the app shows a toast I put into the "notificationReceived" handler and a few seconds later the local notification I triggered for this case.

But only once I unlock the phone it seems.

**Reply by pct-cclausen** [reply, marked as answer]
Nope, it turns out in this case it just takes a while, but it shows. It's good enough for me.
